vision/recognition_service.proto

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2016 <Admobilize>
+ * All rights reserved.
+ */
+
+syntax = "proto3";
+
+package admobilize_vision;
+
+import "vision.proto";
+import "vision_service.proto";
+
+// *********************** StoreDescriptor method  ***************************
+
+message StoreDescriptorRequest {
+
+  // Source images / video to obtain descriptors
+  VisionRequest vision_request = 1;
+
+  // List of descriptors
+  repeated FaceDescriptor face_descriptor = 2;
+
+  // Arbitrary user defined string for the descriptor
+  string tag = 3;
+
+  // Optional device IDs to associate the descriptor
+  // with a set of devices
+  repeated string device_id = 4;
+}
+
+message StoreDescriptorResponse {
+
+  // Stored descriptor IDs in the same order they were sent
+  repeated string uuid = 1;
+
+  // In case store request fails, will contain the reason
+  string reason = 2;
+}
+
+// *********************** GetDescriptor method  ***************************
+
+message GetDescriptorRequest {
+  // Specific descriptors to obtain
+  repeated string uuid = 1;
+
+  // Device ID to get descriptors for
+  string device_id = 2;
+
+  // Next page token when paginating
+  string next_page_token = 3;
+}
+
+message GetDescriptorResponse {
+  // Descriptor list
+  repeated FaceDescriptor face_descriptor = 1;
+
+  // Token use for pagination when there are too many results
+  string next_page_token = 2;
+}
+
+// *********************** DeleteDescriptor method  ***************************
+
+message DeleteDescriptorRequest {
+  // Stored descriptor IDs to delete
+  repeated string uuid = 1;
+}
+
+message DeleteDescriptorResponse {
+  // Amount of currently stored descriptors
+  int32 number_of_descriptors = 1;
+
+  // In case the delete fails, this field will contain the reason
+  string reason = 2;
+}
+
+// *********************** Recognize method  ***************************
+message RecognizeRequest {
+
+  // Source for VisionRequest for recognition
+  VisionRequest vision_request = 1;
+
+  // List of descriptors
+  repeated FaceDescriptor face_descriptor = 2;
+
+  // Signals the desired algorithm for recognitions
+  string matching_algorithm_version = 3;
+}
+
+// Describes descriptor matches
+message DescriptorMatch {
+  // user set tag
+  string tag = 1;
+
+  // Score for the match
+  float score = 2;
+}
+
+message RecognizeResponse {
+  // List of matches 
+  repeated DescriptorMatch recognition_match = 1;
+}
+
+service RecognitionService {
+
+  // Save a descriptor 
+  // The received Recognition request may:
+  //   - contain the calculated descriptors
+  //   - contain a VisionRequest with a populated ImageList field
+  //   - contain a VisionRequest with a populated Video field
+  //
+  // The tag is mandatory for a store call to be successful.
+  // When device_id items are specified, the descriptor will be scoped
+  // to the given device IDs, meaning that match requests will only
+  // succeed when correct device_id and credentials match.
```

Comment:
Is there further validation of device_id? (That the account is the owner of the id?)